R/reduce.R

```diff
@@ -301,12 +301,38 @@ seq_len2 <- function(start, end) {
 #'
 #' @description
 #'
-#' `accumulate()` [reduces][reduce] a vector with a binary function,
-#' keeping all intermediate results, from the initial value to the
-#' final reduced value, i.e. the result you'd have gotten if you used
-#' [reduce()] instead of `accumulate()`.
+#' `accumulate()` sequentially applies a function to elements of a vector taken
+#'  in pairs. The results of each application are returned in a list. The
+#'  accumulation can optionally terminate before processing the whole vector
+#'  in response to a `done()` signal returned by the accumulation function.
```

Comment:
I like the new introduction. Perhaps a second paragraph should mention that accumulate is like reduce but returns all the intermediate results instead of just the final one?